lms/djangoapps/courseware/views.py

```diff
@@ -685,6 +686,10 @@ def course_info(request, course_id):
         staff_access = has_access(request.user, 'staff', course)
         masquerade, user = setup_masquerade(request, course_key, staff_access, reset_masquerade_data=True)
 
+        show_enroll_banner = request.user.is_authenticated() and not CourseEnrollment.is_enrolled(user, course.id)
+        if show_enroll_banner and isinstance(course_key, CCXLocator):
```

Comment:
@singingwolfboy  CCX locator extends Course locator. And the difference is that CCX locator has CCX id (a number) in it